### DISCUSSION TITLE
请问能否把单词翻译也保存下来

### DISCUSSION THREAD

**Discussion by mousepotato** [crimx/ext-saladict#1640] [General]
比如可以默认把 bing 词典的翻译保存下来，否则这个单词本是不完整的。

![image](https://user-images.githubusercontent.com/97764/153105378-4edb434b-ce02-4f6e-ae62-0b940f8b97c0.png)

**Comment by crimx** [comment]
沙拉查词希望引导用户关注上下文从而避免粗粒度收集无用的信息。句子提供了多个翻译结果，单词这个还没想好如何自动化处理，所以先没做了。目前单词是让用户自行挑选符合上下文的释义填进去。

**Reply by mousepotato** [reply]
请问能否告诉我哪里可以做个调整，我希望自己build一个可以自动添加 bing 词典单词的版本。现在的上下文情景，尤其是转到 anki 里面确实不太好用。（有道单词本的导出就不错）

**Reply by crimx** [reply]
你可以在[这里](https://github.com/crimx/ext-saladict/blob/1f95e7584133406dd66bf61c80ff947d979f0bb9/src/content/redux/epics/index.ts#L96) 做异步请求获取需要的信息，修改 word.trans

**Reply by mousepotato** [reply]
您好，我在这里面手动添加的可以被保存到 notebook 里面。我在你提示的 aysnc  (line 96) 处的打印没有被执行。（我不是特别清楚整个流程）。我现在的问题：

- 如果我在下图手动添加的地方如何能够取到 bing 词典的翻译？
- 整个流程（async这段何时会被执行呢）
- 另外不确定我的修改是否是正确的方式，感谢你的时间和帮忙。

<img width="899" alt="image" src="https://user-images.githubusercontent.com/97764/154835389-bbf1fefa-4355-4285-b676-32c0d6f7d3b6.png">

**Reply by crimx** [reply]
> 我在你提示的 aysnc (line 96) 处的打印没有被执行

这个位置保存单词时是必然会执行的。修改代码之后有无重新载入扩展？可以在模块根部打印验证一下是否跑着最新的代码。

> 如果我在下图手动添加的地方如何能够取到 bing 词典的翻译？

如果已经启用了必应词典，可以通过 `state$.value.renderedDicts` 找到已经启用的词典结果，否则你需要自行发请求获取。

> 整个流程（async这段何时会被执行呢）

你是指 96 行那里么？那里是在[收藏时](https://github.com/crimx/ext-saladict/blob/1f95e7584133406dd66bf61c80ff947d979f0bb9/src/content/redux/epics/index.ts#L49)启用了显示单词编辑器`state$.value.config.editOnFav` 才会跑，你不需要显示单词编辑器的话可以删掉让这个方法直接返回一个 Observable 就行，参考[96-116](https://github.com/crimx/ext-saladict/blob/1f95e7584133406dd66bf61c80ff947d979f0bb9/src/content/redux/epics/index.ts#L90-L116)。`from(async () => { /** 你的代码 */ })`。

**Reply by mousepotato** [reply]
谢谢，我通过这样的方法（不是太完美）保存了 bing 词典。

代码：https://github.com/mousepotato/ext-saladict/commit/0e89d24653555348e83eb24c26b0899b7f88f7b1

如果可以改进，请告诉我。再次感谢。

**Reply by mousepotato** [reply]
当然，如果可以让用户配置，是否保存单词（以及可以选择保存哪个词典的单词）就更好了。现在我可以自己编译使用我这个简易版本。
